core/jobs/batch_jobs/test_gcs_io_job.py

```diff
@@ -0,0 +1,167 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests the functionality of gcs_io."""
+
+from __future__ import annotations
+
+from core.jobs import base_jobs
+from core.jobs.io import gcs_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class TestGCSIoWriteJob(base_jobs.JobBase):
+    """Write dummy files to GCS."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        write_files_to_gcs = (
+            self.pipeline
+            | 'Create PCollection having test bucket and test files' >> (
+                beam.Create(
+                    [
+                        {
+                            'filepath': (
+                                'dummy_folder/dummy_subfolder/dummy_file_1'),
+                            'data': b'testing_1'
+                        },
+                        {
+                            'filepath': (
+                                'dummy_folder/dummy_subfolder/dummy_file_2'),
+                            'data': b'testing_2'
+                        }
+                    ]
+                ))
+            | 'Write files to GCS' >> gcs_io.WriteFile()
+        )
+
+        total_files_write = (
+            write_files_to_gcs
+            | 'Total number of files write to GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL FILES WRITTEN'))
+        )
+
+        return total_files_write
+
+
+class TestGCSIoReadJob(base_jobs.JobBase):
+    """Read dummy files from GCS."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        read_files_from_gcs = (
+            self.pipeline
+            | 'Create PCollection of files that needs to be fetched' >> (
+                beam.Create(
+                    [
+                        'dummy_folder/dummy_subfolder/dummy_file_1',
+                        'dummy_folder/dummy_subfolder/dummy_file_2'
+                    ]
+                ))
+            | 'Read files from the GCS' >> gcs_io.ReadFile()
+            | 'Filter the results with OK status' >> beam.Filter(
+                lambda result: result.is_ok())
+            | 'Unwrap the data' >> beam.Map(lambda result: result.unwrap())
```

Comment:
I think this should be only done after line 93 (in `output_files`). Because in `total_files_read` we want to show the errors too.

Reply:
Make sense, I have updated this.